docs: update contribution guidelines and templates (#96)
Women-Devs-SG/womendevssg: .github/PULL_REQUEST_TEMPLATE.md

<!--
❤️ Thank you for your contribution! We appreciate your support.
To help us review this PR efficiently, please read and complete the checklist below.
-->

### 📝 Mandatory Pre-Submission Checklist

**Please check each box. Your Pull Request will not be reviewed if you have not confirmed these steps.**

> - [ ] **I have been assigned this issue.** I commented on the issue and waited for a maintainer to assign it to me before I started working on it.

Reviewer: Nitpick - can be shorter.

```suggestion
- [ ] **I have been assigned this issue.** I commented on the issue and a maintainer has assigned it to me.
```
